src/peft/tuners/lora.py

```diff
@@ -580,15 +579,22 @@ def _unload_and_optionally_merge(self, merge=True, progressbar: bool = False):
 
         return self.model
 
-    def add_weighted_adapter(self, adapters, weights, adapter_name, combination_type="svd"):
+    def add_weighted_adapter(
+        self, adapters, weights, adapter_name, combination_type="svd", svd_rank=None, svd_params={}, svd_clamp=None
+    ):
         """
         This method adds a new adapter by merging the given adapters with the given weights.
 
         Args:
             adapters (list): List of adapter names to be merged.
             weights (list): List of weights for each adapter.
             adapter_name (str): Name of the new adapter.
-            combination_type (str): Type of merging. Can be one of [`svd`, `linear`]
+            combination_type (str): Type of merging. Can be one of [`svd`, `linear`, `cat`]
+            svd_rank (int): Rank of output adapter for svd. If None provided, will use max rank of merging adapters.
```

Comment:
The type annotation is not quite right, as the default is `None`.

Reply:
My bad, fixed the annotation here